### ISSUE TITLE
Split up LaTeX sources using subfiles package

### ISSUE THREAD

**Issue by dietmarw** [modelica/ModelicaSpecification#2297]
When working on #2296 I noticed that the processing time is non-trivial for the whole document. This is sometimes frustrating when only working on one chapter. Luckily there are solutions available and I would suggest to use the `subfiles` packages. 
Some information how this works is given here: https://www.overleaf.com/learn/latex/Multi-file_LaTeX_projects

If OK, I could do this quick conversion and at he same time clean up the structure of the sources a bit. But I don't want to put the effort into this if people dislike the idea.

**Comment by HansOlsson** [comment]
Good idea.

I believe I looked at https://en.wikibooks.org/wiki/LaTeX/Modular_Documents before splitting off the chapters into separate files, and I notice that subfiles is listed as one option there as well.

The minor concerns are:
- The overleaf documentation has "section" before the sub-files. I assume that is just a bad example.
- Image paths for overleaf, I don't know exactly why that would be necessary. But it's anyway just one command setting a path so nothing major.
- Bibliography is listed as an issue, but I don't understand the exact issue - I assume it's only an issue if you want bibliography in both the main document and in the separate chapters, and thus not important at the moment. (For some reason I vaguely recall that having bibliography in both chapters and at end of book was an issue when I wrote my PhD thesis, especially to ensure consistency.)

**Comment by dietmarw** [comment]
> The minor concerns are:
> 
> * The overleaf documentation has "section" before the sub-files. I assume that is just a bad example.

That is just an example but actually doing this way makes it easier to find the correct file from the main file. So if you look for a certain chapter by title you have now in the `MSL.tex` an index for that. 

> * Image paths for overleaf, I don't know exactly why that would be necessary. But it's anyway just one command setting a path so nothing major.

The section files should not sit in the top-level. That just makes it impossible to navigate. So I would adapt the structure that only `MSL.tex` and preamble files etc are on the top level but all sections are in a subfolder. `sections` sounds like a good name to me.

> * Bibliography is listed as an issue, but I don't understand the exact issue - I assume it's only an issue if you want bibliography in both the main document and in the separate chapters, and thus not important at the moment. (For some reason I vaguely recall that having bibliography in both chapters and at end of book was an issue when I wrote my PhD thesis, especially to ensure consistency.)

Yes, probably not an issue.

**Comment by HansOlsson** [comment]
> That is just an example but actually doing this way makes it easier to find the correct file from the main file. So if you look for a certain chapter by title you have now in the MSL.tex an index for that.

True, but as I understand you then don't get the chapter-heading in the sub-files, right?

That causes two concerns:
- The pdf-formatting for the chapter will be a bit off since it starts at a different place on the page.  (When having a file per chapter; for sections neither variant works.)
- You might miss some context at the start of the chapter.

I find that having file-names corresponding to the chapter-names is a reasonable compromise.

> The section files should not sit in the top-level. That just makes it impossible to navigate. So I would adapt the structure that only MSL.tex and preamble files etc are on the top level but all sections are in a subfolder. sections sounds like a good name to me.

Yes, that is good.
However, it looked as if there was some additional issue with image-path. It could just be that I misunderstood it.

**Comment by dietmarw** [comment]
I've created the #2300 for restructuring. It looks latexml was not the best choice when it comes to the subfiles package. Other converters support it. I've filed an issue at https://github.com/brucemiller/LaTeXML/issues/1087

I have already a subfile structure PR ready but need to see if latexml might be able to process it after all or if we have to use the ifpdf package (which would work).

**Comment by dietmarw** [comment]
LaTeXML as of 0.8.4 will have "support" for subfiles(https://github.com/brucemiller/LaTeXML/issues/1087#issuecomment-488726343). The way they integrated it is that LaTeXML does now accept the subfile structure but will not be able to process single sub files. But that is fine since this means that online editors like Overleaf can process single files (and being quicker that way) whilst the HTML build will still work on the main document. 

So as soon as 0.8.4 is available for building I can send my PR for making use of subfiles.

**Comment by beutlich** [comment]
> So as soon as 0.8.4 is available

It's available quite some time now.

**Comment by HansOlsson** [comment]
> > So as soon as 0.8.4 is available
> 
> It's available quite some time now.

Ok, will look into it later this week.

**Comment by dietmarw** [comment]
Back then I had already prepared a branch for this: https://github.com/dietmarw/ModelicaSpecification/tree/sub
It's just one commit but that one needs to get rebased on current master again.

**Comment by HansOlsson** [comment]
> Back then I had already prepared a branch for this: https://github.com/dietmarw/ModelicaSpecification/tree/sub
> It's just one commit but that one needs to get rebased on current master again.

Excellent. Can you do it? I tried and it seemed to create a mess.
